src/xai_grok/grok.py

```diff
@@ -0,0 +1,223 @@
+        response = self._send_request(http.HTTPMethod.GET, endpoint)
+        return self._parse_response_as_type(
+            response, schemas.LanguageModels, endpoint=endpoint
+        )
+
+    def messages(self, request: schemas.MessageRequest) -> schemas.MessageResponse:
+        endpoint = "v1/messages"
+        response = self._send_request(
+            http.HTTPMethod.POST,
+            endpoint,
+            request_body=request.model_dump(exclude_none=True),
+        )
+        if response.status_code == http.client.BAD_REQUEST:
+            raise errors.InvalidRequestError(response.text)
+        if response.status_code == http.client.UNPROCESSABLE_ENTITY:
+            raise errors.IncompleteRequestError(response.text)
+        return self._parse_response_as_type(
+            response, schemas.MessageResponse, endpoint=endpoint
+        )
+
+    def model(self, model_id: str) -> schemas.Model:
+        endpoint = f"v1/models/{model_id}"
+        response = self._send_request(http.HTTPMethod.GET, endpoint)
+        if response.status_code == http.client.NOT_FOUND:
+            raise errors.ModelNotFoundError(f"Model {model_id} not found")
+        return self._parse_response_as_type(response, schemas.Model, endpoint=endpoint)
+
+    def models(self) -> schemas.Models:
+        endpoint = "v1/models"
+        response = self._send_request(http.HTTPMethod.GET, endpoint)
+        return self._parse_response_as_type(response, schemas.Models, endpoint=endpoint)
+
+    def tokenize_text(
+        self, request: schemas.TokenizeTextRequest
+    ) -> schemas.TokenizeTextResponse:
+        endpoint = f"v1/tokenize-text"
+        response = self._send_request(
+            http.HTTPMethod.POST,
+            endpoint,
+            request_body=request.model_dump_json(exclude_none=True),
+        )
+        if response.status_code == http.client.BAD_REQUEST:
+            raise errors.InvalidRequestError(response.text)
+        elif response.status_code == http.client.UNPROCESSABLE_ENTITY:
+            raise errors.IncompleteRequestError(response.text)
+        return self._parse_response_as_type(
+            response, schemas.TokenizeTextResponse, endpoint=endpoint
+        )
+
+    def _send_request(
+        self,
+        method: http.HTTPMethod,
+        endpoint: str,
+        additional_headers: dict = {},
+        request_body: dict = None,
+    ) -> requests.Response:
+        headers = {
+            "Authorization": f"Bearer {self.grok_api_key}",
+            "Content-Type": "application/json",
+        } | additional_headers
+        if method == http.HTTPMethod.GET:
+            response = self.requests.get(
+                f"{self.GROK_API_BASE_URL}/{endpoint}", headers=headers
+            )
+        elif method == http.HTTPMethod.POST:
+            response = self.requests.post(
+                f"{self.GROK_API_BASE_URL}/{endpoint}",
+                headers=headers,
+                json=request_body,
+            )
```

Comment:
The `UnsupportedHttpMethod` exception is raised with an undefined variable `http_method` instead of the correct `method` parameter, which will cause a NameError at runtime.